tracer/src/Datadog.Trace/Activity/OtlpHelpers.cs

```diff
@@ -204,6 +203,95 @@ private static void AgentConvertSpan<TInner>(TInner activity, Span span)
+
+                    var samplingPriority = traceParentSample switch
+                    {
+                        true when traceState.SamplingPriority is > 0 => traceState.SamplingPriority.Value,
+                        true => SamplingPriorityValues.AutoKeep,
+                        false when traceState.SamplingPriority is <= 0 => traceState.SamplingPriority.Value,
+                        false => SamplingPriorityValues.AutoReject,
+                    };
+
+                    var spanContext = new SpanContext(
+                        newActivityTraceId,
+                        newActivitySpanId,
+                        samplingPriority: samplingPriority,
+                        serviceName: null,
+                        origin: traceState.Origin,
+                        isRemote: duckLink.Context.IsRemote);
+
+                    var traceTags = TagPropagation.ParseHeader(traceState.PropagatedTags);
+
+                    if (traceParentSample && traceState.SamplingPriority <= 0)
+                    {
+                        traceTags.SetTag(Tags.Propagated.DecisionMaker, "-0");
+                    }
+                    else if (!traceParentSample && traceState.SamplingPriority > 0)
+                    {
+                        traceTags.RemoveTag(Tags.Propagated.DecisionMaker);
+                    }
+
+                    spanContext.AdditionalW3CTraceState = traceState.AdditionalValues;
+                    spanContext.LastParentId = traceState.LastParent;
+                    spanContext.PropagatedTags = traceTags;
+
+                    var extractedSpan = new Span(spanContext, DateTimeOffset.Now, new CommonTags());
+                    var spanLink = span.AddSpanLink(extractedSpan);
+
+                    if (duckLink.Tags is not null)
+                    {
+                        foreach (var kvp in duckLink.Tags)
+                        {
+                            if (!string.IsNullOrEmpty(kvp.Key)
+                             && IsAllowedAtributeType(kvp.Value))
+                            {
+                                if (kvp.Value is Array array)
+                                {
+                                    int index = 0;
+                                    foreach (var item in array)
+                                    {
+                                        if (item is not null)
+                                        {
+                                            spanLink.AddAttribute($"{kvp.Key}.{index}", item.ToString()!);
+                                            index++;
+                                        }
+                                    }
+                                }
+                                else
+                                {
+                                    spanLink.AddAttribute(kvp.Key, kvp.Value!.ToString()!);
+                                }
```

Comment:
I think this should probably still guard against null?
```suggestion
                                else if (kvp.Value?.ToString() is { } kvpValue)
                                {
                                    spanLink.AddAttribute(kvp.Key, kvpValue);
                                }
```